mindsdb/integrations/handlers/confluence_handler/confluence_table.py

```diff
@@ -0,0 +1,119 @@
+import pandas as pd
+
+from typing import List
+
+from mindsdb.integrations.libs.api_handler import APITable
+from mindsdb.integrations.utilities.sql_utils import extract_comparison_conditions
+from mindsdb.utilities.log import get_log
+
+from mindsdb_sql.parser import ast
+
+logger = get_log("integrations.confluence_handler")
+
+class ConfluenceSpacesTable(APITable):
+    """Confluence Spaces Table implementation"""
+
+    def select(self, query: ast.Select) -> pd.DataFrame:
+        """Pulls data from the Confluence "get_all_spaces" API endpoint
+        Parameters
+        ----------
+        query : ast.Select
+           Given SQL SELECT query
+        Returns
+        -------
+        pd.DataFrame
+            confluence "get_all_spaces" matching the query
+        Raises
+        ------
+        ValueError
+            If the query contains an unsupported condition
+        """
+        conditions = extract_comparison_conditions(query.where)
+
+        if query.limit:
+            total_results = query.limit.value
+        else:
+            total_results = 50
+
+
+        spaces_kwargs = {}
+        order_by_conditions = {}
+
+        if query.order_by and len(query.order_by) > 0:
+            order_by_conditions["columns"] = []
+            order_by_conditions["ascending"] = []
+
+            for an_order in query.order_by:
+                if an_order.field.parts[0] != "":
+                    next    
+                if an_order.field.parts[1] in self.get_columns():
+                    order_by_conditions["columns"].append(an_order.field.parts[1])
+
+                    if an_order.direction == "ASC":
+                        order_by_conditions["ascending"].append(True)
+                    else:
+                        order_by_conditions["ascending"].append(False)
+                else:
+                    raise ValueError(
+                        f"Order by unknown column {an_order.field.parts[1]}"
+                    )
+
+        for a_where in conditions:
+            if a_where[1] == "type":
+                if a_where[0] != "=":
+                    raise ValueError("Unsupported where operation for type")
+                if a_where[2] not in ["personal", "global"]:
+                    raise ValueError(
+                        f"Unsupported where argument for state {a_where[2]}"
+                    )
+                spaces_kwargs["type"] = a_where[2]
+            else:
+                raise ValueError(f"Unsupported where argument {a_where[1]}")
+
+        confluence_spaces_df = pd.json_normalize(self.handler.connect().get_all_spaces(start=0,limit=total_results)["results"])
```

Comment:
Can we make this to be two lines since it will fails on pep8 validation?

Reply:
Fixed this, thanks for pointing it out, zoran